src/app/components/transactionSetting/editBtcFee.tsx

```diff
@@ -0,0 +1,421 @@
+}
+function EditBtcFee({
+  type,
+  fee,
+  feeRate,
+  btcRecipients,
+  ordinalTxUtxo,
+  isRestoreFlow,
+  nonOrdinalUtxos,
+  feeMode,
+  error,
+  customFeeSelected,
+  setIsLoading,
+  setIsNotLoading,
+  setFee,
+  setFeeRate,
+  setError,
+  setFeeMode,
+  setCustomFeeSelected,
+  feeOptionSelected,
+}: Props) {
+  const { t } = useTranslation('translation');
+  const { network, btcAddress, btcFiatRate, fiatCurrency, selectedAccount, ordinalsAddress } =
+    useWalletSelector();
+  const [totalFee, setTotalFee] = useState(fee);
+  const [feeRateInput, setFeeRateInput] = useState(feeRate?.toString() ?? '');
+  const inputRef = useRef<HTMLInputElement>(null);
+  const debouncedFeeRateInput = useDebounce(feeRateInput, 500);
+  const { ordinals } = useOrdinalsByAddress(btcAddress);
+  const ordinalsUtxos = useMemo(() => ordinals?.map((ord) => ord.utxo), [ordinals]);
+  const btcClient = useBtcClient();
+  const feeData = useBtcFees({
+    isRestoreFlow: !!isRestoreFlow,
+    nonOrdinalUtxos,
+    btcRecipients,
+    type,
+    ordinalTxUtxo,
+  });
+
+  useEffect(() => {
+    setFee(totalFee);
+  }, [totalFee]);
+
+  const recalculateFees = async () => {
+    if (type === 'BTC') {
+      try {
+        setIsLoading();
+        setError('');
+
+        if (isRestoreFlow) {
+          const { fee: modifiedFee, selectedFeeRate } = await getBtcFeesForNonOrdinalBtcSend(
+            btcAddress,
+            nonOrdinalUtxos!,
+            ordinalsAddress,
+            network.type,
+            feeMode,
+            feeRateInput,
+          );
+          setFeeRateInput(selectedFeeRate!.toString());
+          setTotalFee(modifiedFee.toString());
+        } else if (btcRecipients && selectedAccount) {
+          const { fee: modifiedFee, selectedFeeRate } = await getBtcFees(
+            btcRecipients,
+            btcAddress,
+            btcClient,
+            network.type,
+            feeMode,
+            feeRateInput,
+          );
+          setFeeRateInput(selectedFeeRate!.toString());
+          setTotalFee(modifiedFee.toString());
+        }
+      } catch (err: any) {
+        if (Number(err) === ErrorCodes.InSufficientBalance) {
+          setError(t('TX_ERRORS.INSUFFICIENT_BALANCE'));
+        } else if (Number(err) === ErrorCodes.InSufficientBalanceWithTxFee) {
+          setError(t('TX_ERRORS.INSUFFICIENT_BALANCE_FEES'));
+        } else setError(err.toString());
+      } finally {
+        setIsNotLoading();
+      }
+    } else if (type === 'Ordinals' && btcRecipients && ordinalTxUtxo) {
+      try {
+        setIsLoading();
+        setError('');
+
+        const { fee: modifiedFee, selectedFeeRate } = await getBtcFeesForOrdinalSend(
+          btcRecipients[0].address,
+          ordinalTxUtxo,
+          btcAddress,
+          btcClient,
+          network.type,
+          ordinalsUtxos || [],
+          feeMode,
+          feeRateInput,
+        );
+        setFeeRateInput(selectedFeeRate!.toString());
```

Comment:
Let's add a check for the `selectedFeeRate` to get rid of `!` symbol here to make the code safer